datasets/hendrycks_test/hendrycks_test.py

```diff
@@ -0,0 +1,188 @@
+        features = datasets.Features(
+            {
+                "question": datasets.Value("string"),
+                "choices": datasets.features.Sequence(datasets.Value("string")),
+                "answer": datasets.features.ClassLabel(num_classes=4, names=["A", "B", "C", "D"]),
+            }
+        )
+        return datasets.DatasetInfo(
+            # This is the description that will appear on the datasets page.
+            description=_DESCRIPTION,
+            # This defines the different columns of the dataset and their types
+            features=features,  # Here we define them above because they are different between the two configurations
+            # If there's a common (input, target) tuple from the features,
+            # specify them here. They'll be used if as_supervised=True in
+            # builder.as_dataset.
+            supervised_keys=None,
+            # Homepage of the dataset for documentation
+            homepage=_HOMEPAGE,
+            # Citation for the dataset
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        data_dir = dl_manager.download_and_extract(_URL)
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split("auxiliary_train"),
+                # These kwargs will be passed to _generate_examples
+                gen_kwargs={
+                    "datadir": os.path.join(data_dir, "data", "auxiliary_train"),
+                    "split": "auxiliary_train",
+                },
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.TEST,
+                # These kwargs will be passed to _generate_examples
+                gen_kwargs={"datadir": os.path.join(data_dir, "data", "test"), "split": "test"},
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.VALIDATION,
+                # These kwargs will be passed to _generate_examples
+                gen_kwargs={
+                    "datadir": os.path.join(data_dir, "data", "val"),
+                    "split": "val",
+                },
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split("dev"),
+                # These kwargs will be passed to _generate_examples
+                gen_kwargs={
+                    "datadir": os.path.join(data_dir, "data", "dev"),
+                    "split": "dev",
+                },
+            ),
+        ]
+
+    def _generate_examples(self, datadir, split):
+        """Yields examples as (key, example) tuples."""
+        # This method handles input defined in _split_generators to yield (key, example) tuples from the dataset.
+        # The `key` is here for legacy reason (tfds) and is not important in itself.
+
+        id_ = 0
+        if split == "auxiliary_train":
+            for f in os.listdir(datadir):
```

Comment:
we need to make the order deterministic:
```suggestion
            for f in sorted(os.listdir(datadir)):
```